runtime/manifest.js

```diff
@@ -877,6 +877,7 @@ ${e.message}
           const hostedParticleLiteral = hostedParticle.clone().toLiteral();
           let particleSpecHash = await digest(JSON.stringify(hostedParticleLiteral));
           let id = `${manifest.generateID()}:${particleSpecHash}:${hostedParticle.name}`;
+          hostedParticleLiteral.id = id;
```

Comment:
We didn't have this and we had to add it for some reason, either a comment or a test or both?

Reply:
A comment inline there would be weird. Everything we store should have an ID. We got away with not having this for stored particles before but once they can be stored by reference then a bunch of tests break if they don't have one.